### ISSUE TITLE
openapi-react-query: Add the ability to bring queryClient as argument

### ISSUE THREAD

**Issue by iffa** [openapi-ts/openapi-typescript#1805]
Noticed that this wasn't possible yet and was a TODO in the code too.

**Comment by kerwanp** [comment]
Temporarily, it is possible to wrap inside a `QueryClientProvider`. But in the future we should be able to pass the queryClient as an argument.

@tanstack/react-query
```tsx
const query = useQuery({
  queryKey: [],
  queryFn: () => {},
}, queryClient);
```

openapi-react-query
```tsx
const query = $api.useQuery('get', '/api/posts', {}, queryClient);
```

Feel free to open a pull request! This should be pretty straight forward.

**Comment by kerwanp** [comment]
Closing this issue as this is resolved in #1814 thanks to @yoshi2no !
